src/fmelt.c

```diff
@@ -618,7 +618,7 @@ SEXP getidcols(SEXP DT, SEXP dtnames, Rboolean verbose, struct processData *data
         // From assign.c's memcrecycle - only one SET_STRING_ELT per RHS item is needed to set generations (overhead)
         for (k=0; k<data->nrow; k++) SET_STRING_ELT(target, k, STRING_ELT(thiscol, k));
         for (j=1; j<data->lmax; j++)
-          memcpy((char *)DATAPTR(target)+j*data->nrow*size, (char *)DATAPTR(target), data->nrow*size);
+          memcpy((char *)CHAR(target)+j*data->nrow*size, (char *)CHAR(target), data->nrow*size);
```

Comment:
and here `CHAR` macro

Reply:
this should stress this change in `getidcols` fun, tested on master vs this branch and looks good
```r
d = data.table(a=as.character(letters[1:5]), v1=1:5, v2=1:5)
d[, "b" := paste0(a,rev(a),a)]
d[, "a" := paste0(a,a,a)]
melt.data.table(d, id=c("a","b"), measure.vars=c("v1","v2"))
```